### PR TITLE
Use output-field field in inference requests 

### LINKED ISSUE #622: [BUG]: Inference job output field is ignored
### Description

When a csv with an (empty or populated) ground_truth column is uploaded (regardless of the name) and we launch an inference job on it, setting the desired output_field (i.e. the field we want predictions to be added to), the service succeeds but promptly ignores output_field and still calls the field `predictions`.

### Reproduction

1. Upload file with empty gt
2. Repeat step 2 above, but with this new dataset.
3. See job succeed. 
4. Use /job/job_id/results/download to get the uri to a file like the one attached.

See:

* input file with empty gt column: [dialogsum_mini_empty_gt.csv](https://github.com/user-attachments/files/18354141/dialogsum_mini_empty_gt.csv)
* output file for the input with empty gt column: [results_empty_gt_requested_gt_field.json](https://github.com/user-attachments/files/18354150/results_empty_gt_requested_gt_field.json)

### Relevant log output

```shell
See above:

* input file with empty gt column
* output file for the input with empty gt column
```

### Expected behavior

Obtaining a file where, if the output_field requested is "model_prediction", for example, there is a new column named "model_prediction".

### System Info

* OS and LInux
* Current main: [34873f6](https://github.com/mozilla-ai/lumigator/commit/34873f680f7d944610c65b077d6a44736607246c)

### Have you searched for similar issues before submitting this one?

- [x] Yes, I have searched for similar issues

### PR DESCRIPTION
**Pull request by javiermtorres**
# What's changing

The default inference config template was not passing the `output_field` to the inference job, so the ground truth column was not being generated. Now this key is passed, with a default value of `predictions` to match the inference job expectations.

The path for the job logs checks in the integration tests is also fixed.

Closes #622 

# How to test it

A new test integration case `test_upload_data_no_gt_launch_annotation` has been added.

# Additional notes for reviewers

N/A

# I already...

- [x] Tested the changes in a working environment to ensure they work as expected
- [x] Added some tests for any new functionality
- [ ] Updated the documentation (both comments in code and [product documentation](https://mozilla-ai.github.io/lumigator) under `/docs`)
- [x] Checked if a (backend) DB migration step was required and included it if required
  - No changes in the DB needed

### CODE REVIEW

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py [hunk 1]
```diff
@@ -78,7 +83,7 @@ def test_upload_data_launch_job(
     assert succeeded
 
     logs_evaluation_job_response = local_client.get(
-        f"/health/jobs/{create_evaluation_job_response_model.id}/logs"
+        f"/jobs/{create_evaluation_job_response_model.id}/logs"
```
Comment (ividal) [inline lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:86]: 🤦‍♀️ How was this not failing earlier?
/me worries

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py [hunk 2]
```diff
@@ -123,13 +128,89 @@ def test_upload_data_launch_job(
...
+    }
+
+    annotation_payload = {
+        "name": "test_annotate",
+        "description": "Test run for Huggingface model",
+        "dataset": str(created_dataset.id),
+        "max_samples": 2,
+        "task": "summarization",
+    }
+    create_inference_job_response = local_client.post(
```
Comment (ividal) [inline lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:167]: Technically create_annotate(...). I know under the hood it's inference, but it's just to mentally keep track of what schema we should be following. 

For example, if we set the model here, it should be rejected. If we set it in a real, full inference job, it should go through.

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py (hunk 1 above)
Comment (javiermtorres) [reply lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:86]: Because there's no failing call :-/ It will say "404, {}" and everything is fine :)

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py (hunk 2 above)
Comment (javiermtorres) [reply lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:167]: Sorry, it is changed to inference in further vars :( Fixing the name 👍 

> For example, if we set the model here, it should be rejected. If we set it in a real, full inference job, it should go through.

This is the basis for a new unit test....

**Review by ividal — APPROVED** [review]
Thanks 🙏 !
Pre-approved and just a small comment on naming.

File: lumigator/python/mzai/backend/backend/services/jobs.py [hunk 3]
```diff
@@ -201,6 +201,9 @@ def create_job(
             name=request.name, description=request.description, experiment_id=experiment_id
         )
 
+        if isinstance(request, JobInferenceCreate) and not request.output_field:
```
Comment (aittalam) [inline lumigator/python/mzai/backend/backend/services/jobs.py:204]: if the request is of type `JobInferenceCreate` it should always have an output field: https://github.com/mozilla-ai/lumigator/blob/906ed4531d5cfe9d1d554694e957b33d0526ed58/lumigator/python/mzai/schemas/lumigator_schemas/jobs.py#L81

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py [hunk 4]
```diff
@@ -123,13 +128,89 @@ def test_upload_data_launch_job(
...
+        get_job_response = local_client.get(f"/jobs/{create_annotation_job_response_model.id}")
+        assert get_job_response.status_code == 200
+        get_job_response_model = JobResponse.model_validate(get_job_response.json())
+        if get_job_response_model.status == JobStatus.SUCCEEDED.value:
+            succeeded = True
+            break
+        if get_job_response_model.status == JobStatus.FAILED.value:
+            succeeded = False
+            break
+        time.sleep(10)
```
Comment (aittalam) [inline lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:187]: We'll now have the same for loop replicated 4 times in this file. Could you please factor out the common code into a function and call that instead?
Also, 200*10 = 2000 seconds for each of these tests is definitely more than anyone is willing to wait. We should cap this to a reasonable amount of time and fail the test otherwise, warning that it failed due to the time cap so we can improve the way we are testing.

File: lumigator/python/mzai/backend/backend/services/jobs.py (hunk 3 above)
Comment (javiermtorres) [reply lumigator/python/mzai/backend/backend/services/jobs.py:204]: AFAICT both `None`s and `""`s are (quite conveniently) falses:

```python
>>> somestr = None
>>> if somestr:
...   print("yay")
... else:
...   print("nope")
... 
nope
>>> somestr = ""
>>> if somestr:
...   print("yay")
... else:
...   print("nope")
... 
nope
>>> somestr = "blurb"
>>> if somestr:
...   print("yay")
... else:
...   print("nope")
... 
yay
``` 

If you can think of other invalid values we can check those explicitly.

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py (hunk 4 above)
Comment (javiermtorres) [reply lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:187]: Yes, now that it's working and we have a clear picture of how the tests should go, we can refactor the integration tests, also reducing the waiting time. Maybe we can make it a separate PR, but if you prefer I'll include it in this one.

Is there some specific annoying corner case related to this waiting time that you'd like to be addressed?

File: lumigator/python/mzai/backend/backend/services/jobs.py (hunk 3 above)
Comment (aittalam) [reply lumigator/python/mzai/backend/backend/services/jobs.py:204]: Whoops sorry, just gave for granted that "" was not false, should have checked 😅

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py (hunk 4 above)
Comment (aittalam) [reply lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py:187]: I think it's ok to have an issue for that! I have opened [this](https://github.com/mozilla-ai/lumigator/issues/643) so you are not blocked on this. Feel free to merge after SDK tests are addressed!

File: lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py (hunk 2 above)
Comment (javiermtorres) [reply lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py]: Ok, clarification. There's no need for a new unit test since the `JobAnnotationCreate` model does not allow a `model` field anyway (it would be dropped). The var names have been updated.
